src/main/java/org/jabref/gui/JabRefFrame.java

```diff
@@ -1317,18 +1317,17 @@ private boolean confirmClose(BasePanel panel) {
             // The user wants to save.
             try {
                 SaveDatabaseAction saveAction = new SaveDatabaseAction(panel, Globals.prefs);
-                if (!saveAction.save()) {
-                    // The action was either canceled or unsuccessful.
-                    output(Localization.lang("Unable to save library"));
-                    return false;
+                if (saveAction.save()) {
+                    // Saved, now exit.
+                    return true;
                 }
+                // The action was either canceled or unsuccessful.
+                output(Localization.lang("Unable to save library"));
```

Comment:
If I understand the code correctly, then even if the save action was not successful (either by returning false or throwing an exception), then JabRef is nonetheless closed. This is not desired, because we want to give the user the chance to rescue her work before closing.

Reply:
Of-course, my bad. I have added a return false statement in case an exception occurs or the save fails.